path_service_principal_test.go

```diff
@@ -700,3 +674,17 @@ func TestCredentialInteg(t *testing.T) {
 		}
 	})
 }
+
+// fakeSaveLoad will simulate the JSON encoding/decoding process that secrets will normally go
+// through. If not done, some of the secret data will retain richer data types (e.g. pointers)
+// than would normally be seen.
+func fakeSaveLoad(s *logical.Secret) {
+	enc, err := jsonutil.EncodeJSON(s)
+	if err != nil {
+		panic(err)
```

Comment:
Is `panic` necessary here?  Should we just `t.error()` instead?

Reply:
Since this is well into "should never happen" (within the context of a test) territory, I'm ok with a loud failure and test halt. That said, I just discovered I have a helper file with a number of these so I moved it over there.